src/storage/secondary/rowset/mem_rowset.rs

```diff
@@ -134,7 +136,8 @@ impl SecondaryMemRowset<BTreeMapMemTable> {
         let chunk = self.mem_table.flush()?;
         let mut builder = self.rowset_builder;
         builder.append(chunk);
-        builder.finish_and_flush().await?;
+        let writer = RowsetWriter::new(self.directory);
```

Comment:
Pass `directory` as a parameter here? So that some upper logic should be adjusted as well.